figures.go

```diff
@@ -0,0 +1,84 @@
+/*        figures.go         */
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Figure interface {
+	area() float64
+	perimeter() float64
+}
+
+type Square struct {
+	x float64
+}
+
+func (s Square) area() float64 {
+	return s.x * s.x
+}
+
+func (s Square) perimeter() float64 {
+	return s.x * 4
+}
+
+type Circle struct {
+	r float64
+}
+
+func (c Circle) area() float64 {
+	z := math.Pi * math.Pow(c.r, 2)
+	return z
+}
+
+func (c Circle) perimeter() float64 {
+	z := math.Pi * 2 * c.r
+	return z
+}
+
+func main() {
+	var s Figure = Square{}
```

Comment:
Just an advice. To get a deeper understanding of `interface` concept, try to add constructor(or fabrics...) functions, like 
```go
func NewCircle(radius float64) (Figure, error){}
```
cuz atm you can safely drop the `Figure` interface from code w/o losing any logic

Reply:
@wshaman 
Understood, but in the assignment the function was given like that

Reply:
Right, sorry. I've missed this description.